@here/harp-mapview/lib/SphereHorizon.ts

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EarthConstants, MathUtils, Projection, ProjectionType } from "@here/harp-geoutils";
+import { assert, Math2D } from "@here/harp-utils";
+import * as THREE from "three";
+
+import { MapViewUtils } from "./Utils";
+
+const twoPi = Math.PI * 2;
+
+// Keep counter clockwise order.
+export enum CanvasSide {
+    Bottom,
+    Right,
+    Top,
+    Left
+}
+
+/**
+ * Class computing horizon tangent points and intersections with canvas for spherical projection.
+ *
+ * @remarks
+ *
+ * The horizon for a sphere is a circle on a plane with a normal matching the sphere normal at the
+ * camera position, and its center is at the interesection of the camera position vector with that
+ * plane. This circle is formed by all intersection points on the sphere with all tangents passing
+ * through the current camera position.
+ *
+//                                            ::::::::::::::::::*`
```

Comment:
Awesome ASCII art, but to be honest I cannot see anything. Could also be the Death Star :smile: 
Alternatively you could just add a svg file to the repository (e.g. in docs/@docs) and add the link to the raw file (that will only be available after merging)

Reply:
I already created some svg when working on the rotation around arbitrary screen point:
https://raw.githubusercontent.com/heremaps/harp.gl/f1e7f4c55e7c472048baaf97639ab8fe185be661/docs/%40docs/globe-math/globe-tilt.svg
I was using Inkscape to create it. If you want I can also share the source file.

Reply:
thanks for the review, but you jumped in slightly too soon and I didn't have the documentation completed. I was toying with an app that converts diagrams to ascii art, hence that uninformative ascii blob. I'm tweaking it right now and adding text so that's useful.